### PR TITLE
[Feature #786] Introduce Scalafmt to auto format scala code

### LINKED ISSUE #786: [Feature] Introduce Scalafmt to auto format scala code
### Search before asking

- [X] I had searched in the [feature](https://github.com/InterestingLab/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Introduce Scalafmt to auto format scala code

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by simon824**
## Purpose of this pull request
Introduce Scalafmt to auto format scala code
closes #786 

#787 got too many conflicts, hard to fix, so push a new pr.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

**Comment by simon824** [comment]
cc @zhongjiajie @CalvinKirs

File: seatunnel-config/src/test/resources/equiv01/no-whitespace.json [hunk 1]
```diff
@@ -1 +1,49 @@
...
+      1,
+      2
+    ],
+    "123": [
+      1,
+      2,
+      3
+    ],
+    "ofString": [
+      "a",
```
Comment (leo65535) [inline seatunnel-config/src/test/resources/equiv01/no-whitespace.json:34]: just a nit: the following form is friendly.
```
"123": [1, 2, 3]
```

**Comment by leo65535** [comment]
+1, Most formats are good, but few formats are unfriendly.

File: seatunnel-config/src/test/resources/equiv01/no-whitespace.json (hunk 1 above)
Comment (simon824) [reply seatunnel-config/src/test/resources/equiv01/no-whitespace.json]: fixed

**Comment by zhongjiajie** [comment]
> 

@leo65535  Could you get an example? and do you have any suggestions about it?

**Comment by simon824** [comment]
> @leo65535 Could you get an example? and do you have any suggestions about it?  

image
https://github.com/InterestingLab/seatunnel/pull/806#discussion_r770301621
Has excluded `json` file format.

**Comment by kezhenxu94** [comment]
I remember we divided to migrate from Scala to Java? If so, I think no new Scala codes will be merged and existing Scala codes will be migrated gradually. Do you think we need to format the Scala codes at this time?

**Comment by simon824** [comment]
> I remember we divided to migrate from Scala to Java? If so, I think no new Scala codes will be merged and existing Scala codes will be migrated gradually. Do you think we need to format the Scala codes at this time?

We have disscussed it before. 
https://github.com/InterestingLab/seatunnel/pull/787#issuecomment-994935568 cc @kezhenxu94

**Comment by kezhenxu94** [comment]
> > I remember we divided to migrate from Scala to Java? If so, I think no new Scala codes will be merged and existing Scala codes will be migrated gradually. Do you think we need to format the Scala codes at this time?
> 
> 
> 
> We have disscussed it before. 
> 
> https://github.com/InterestingLab/seatunnel/pull/787#issuecomment-994935568 cc @kezhenxu94 
> 
> 
> 
> 

Ok. Make sense.

**Review by CalvinKirs — COMMENTED** [review]
It needs to be declared in root-pom so that our CI will check

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**